R/pkg/R/functions.R

```diff
@@ -2163,10 +2164,15 @@ setMethod("date_format", signature(y = "Column", x = "character"),
 #' df2 <- sql("SELECT named_struct('name', 'Bob') as people")
 #' df2 <- mutate(df2, people_json = to_json(df2$people))
 #' schema <- structType(structField("name", "string"))
-#' head(select(df2, from_json(df2$people_json, schema)))}
+#' head(select(df2, from_json(df2$people_json, schema)))
+#' head(select(df2, from_json(df2$people_json, "name STRING")))}
 #' @note from_json since 2.2.0
-setMethod("from_json", signature(x = "Column", schema = "structType"),
+setMethod("from_json", signature(x = "Column", schema = "characterOrstructType"),
           function(x, schema, as.json.array = FALSE, ...) {
+            if (is.character(schema)) {
+              schema <- structTypeFromDDL(schema)
+            }
```

Comment:
I see. So, you mean ...
`structTypeFromDDL` ->`structType.character` for `structType("a STRING, b INT")` usage?

This makes much more sense to me and I'd like to propose `StructType(a STRING, b INT")` case in Scala/Python side too.

For me, I would like to avoid JSON format if DDL format is allowed in any event for now although we should keep our eyes on mailing list and JIRAs.